File: pop.m
#include "monty.h"

/**
 * pop - Removes the top element .
 * @stack: Pointer tothe stack.
 * @line_numm: Line num in the file where the opcode is located.
 */
void pop(stack_t **stack, unsigned int line_numm)
{
    stack_t *top;

    if (*stack == NULL)
    {
        fprintf(stderr, "L%u: can't pop an empty stack\n", line_numm);
        exit(EXIT_FAILURE);
    }

    top = *stack;
    *stack = top->next;
    if (*stack != NULL)
        (*stack)->prev = NULL;
    free(top);
}

File: pop.c
#include "monty.h"

/**
 * f_pop - pr!nts th£ t0p
 * @head: st@ck h£@d
 * @counter: !in£_num
 * Return: n0th!ng
 */
void f_pop(stack_t **head, unsigned int counter)
{
	stack_t *h;

	if (*head == NULL)
	{
		fprintf(stderr, "L%d: can't pop an empty stack\n", counter);
		fclose(bus.file);
		free(bus.content);
		free_stack(*head);
		exit(EXIT_FAILURE);
	}
	h = *head;
	*head = h->next;
	free(h);
}

